osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -475,6 +475,21 @@ protected void ApplyResult(Action<JudgementResult> application)
             if (!Result.HasResult)
                 throw new InvalidOperationException($"{GetType().ReadableName()} applied a {nameof(JudgementResult)} but did not update {nameof(JudgementResult.Type)}.");
 
+            // Some (especially older) rulesets use scorable judgements instead of the newer ignorehit/ignoremiss judgements.
```

Comment:
Will this code be removable with the obsoletion of the methods marked as such in this PR, or does it have to remain here for now?

Reply:
In practice this could lead implementers down the wrong path of doing this as a natural thing so I want to remove it somehow.

The validation check takes care of everything just below this, but I’m not sure how to signal correct usage to the user while still allowing current code... Maybe obsoletion notice via important log entries?

Reply:
I've added an obsoletion notice via logger messages.